### PR TITLE
INT-1539 - Only pull issues from projects that are specified in the config

### PR DESCRIPTION
**Pull request by mknoedel**
Solves this issue:
image

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -136,12 +136,26 @@ export class APIClient {
...
+          pageSize: ISSUES_PAGE_SIZE,
+        });
+      } catch (err: any) {
+        if (err?.message.includes(`The value '${projectKey}' does not exist for the field 'project'.`)) {
+          this.logger.info(
+            { projectKey },
+            'Project key does not exist or you do not have access to pull down issues from this project.',
+          );
+          break
+          // This error is fine, just break from the loop
```
Comment (mknoedel) [inline src/client.ts:154]: Handle specific error and move on.

File: src/steps/issues.ts [hunk 2]
```diff
@@ -34,6 +34,8 @@ export async function fetchIssues({
   const config = instance.config;
   const apiClient = createAPIClient(config, logger);
 
+  const projectConfigs = buildProjectConfigs(instance.config.projects);
+
```
Comment (mknoedel) [inline src/steps/issues.ts:39]: This is the main change. We only want to ingest issues from projects that are specified in the config. `buildProjectConfigs` helps sanitize the config.

File: src/steps/issues.ts [hunk 3]
```diff
@@ -71,13 +73,23 @@ export async function fetchIssues({
...
         )) as IssueEntity;
 
-        await jobState.addRelationship(
-          createDirectRelationship({
-            _class: RelationshipClass.HAS,
-            from: projectEntity,
-            to: issueEntity,
-          }),
+        const projectEntity = projectEntities?.find(
+          (project) => project.key === projectConfig.key,
```
Comment (mknoedel) [inline src/steps/issues.ts:78]: This isn't the most efficient way to find the project entity key, but this array is going to be very small so it should be fine.

File: src/steps/projects.ts [hunk 4]
```diff
@@ -1,24 +1,24 @@
 import {
```
Comment (mknoedel) [inline src/steps/projects.ts:1]: This file was formatted with prettier. No other changes.

**Review by ceelias — APPROVED** [review]

**Review by ceelias — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,15 +6,21 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to
 [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased](https://github.com/JuptiterOne/graph-jira/compare/v1.9.8...HEAD)
+
+## [2.1.0] - 2021-06-10
```
Comment (ceelias) [inline CHANGELOG.md:11]: Date is wrong

**Pull request merged**